### PR TITLE
feat: add keys endpoint

### PR DESCRIPTION
**Pull request by benjamb**
While this endpoint may be a little niche, it was useful for a project I was involved in recently.

### CODE REVIEW

**Comment by JohnVillalovos** [comment]
> On a somewhat unrelated note, is there a reason flake8 isn't included in the pre-commit config?

Good catch. @nejch recently updated it to add `isort` and `black`. But likely we should add `flake8' and `mypy``to it too.

File: gitlab/v4/objects/keys.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    pass
+
+
+class KeyManager(GetMixin, RESTManager):
+    _path = "/keys"
+    _obj_cls = Key
+
+    def get(self, id=None, **kwargs):
+        if id:
+            return super(KeyManager, self).get(id, kwargs)
```
Comment (JohnVillalovos) [inline gitlab/v4/objects/keys.py:20]:
```suggestion
        if id is not None:
            return super(KeyManager, self).get(id, **kwargs)
```

**Review by JohnVillalovos — CHANGES_REQUESTED** [review]

File: gitlab/v4/objects/keys.py (hunk 1 above)
Comment (benjamb) [reply gitlab/v4/objects/keys.py:20]: Erk, good spot.

Comment (benjamb) [reply gitlab/v4/objects/keys.py]: @JohnVillalovos I've squashed this in.

**Review by JohnVillalovos — APPROVED** [review]
LGTM. Thanks!

I'd like @nejch and/or @max-wittig  to also look.

File: gitlab/v4/objects/keys.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+class KeyManager(GetMixin, RESTManager):
+    _path = "/keys"
+    _obj_cls = Key
+
+    def get(self, id=None, **kwargs):
+        if id is not None:
+            return super(KeyManager, self).get(id, **kwargs)
+
+        if "fingerprint" not in kwargs:
+            raise AttributeError("Missing attribute: fingerprint")
```
Comment (nejch) [inline gitlab/v4/objects/keys.py:23]: This should probably have both options:

```suggestion
            raise AttributeError("Missing attribute: id or fingerprint")
```

File: docs/gl_objects/keys.rst [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+Reference
+---------
+
+* v4 API
+
+  + :class:`gitlab.v4.objects.Key`
+  + :class:`gitlab.v4.objects.KeyManager`
+  + :attr:`gitlab.Gitlab.keys`
+
+* GitLab API: https://docs.gitlab.com/ce/api/keys.html
```
Comment (nejch) [inline docs/gl_objects/keys.rst:17]: Can we have just a quick note that for SSH keys, this requires admin access? Like here:
https://github.com/python-gitlab/python-gitlab/blob/eabe091945d3fe50472059431e599117165a815a/docs/gl_objects/variables.rst#L12

**Review by nejch — CHANGES_REQUESTED** [review]
Thanks a lot, looks great! Just 2 tiny comments :)

**Comment by nejch** [comment]
> > On a somewhat unrelated note, is there a reason flake8 isn't included in the pre-commit config?
> 
> Good catch. @nejch recently updated it to add `isort` and `black`. But likely we should add `flake8` and `mypy` to it too.

Yes, these were just very recent additions to CI checks but we should add pre-commit as well.

File: gitlab/v4/objects/keys.py (hunk 2 above)
Comment (benjamb) [reply gitlab/v4/objects/keys.py]: Done.

File: docs/gl_objects/keys.rst (hunk 3 above)
Comment (benjamb) [reply docs/gl_objects/keys.rst:17]: And done.

**Review by nejch — APPROVED** [review]

**Pull request merged**